demo/admin/src/products/ProductsGrid.tsx

```diff
@@ -54,20 +54,20 @@ function ProductsGridToolbar() {
             <ToolbarItem>
                 <GridColumnsButton />
             </ToolbarItem>
-            <ToolbarItem>
-                <Button startIcon={<AddIcon />} component={StackLink} pageName="add" payload="add" variant="contained" color="primary">
-                    <FormattedMessage id="products.newProduct" defaultMessage="New Product" />
-                </Button>
-            </ToolbarItem>
+            {toolbarAction && <ToolbarActions>{toolbarAction}</ToolbarActions>}
         </DataGridToolbar>
     );
 }
 
 type Props = {
     filter?: GQLProductFilter;
+    toolbarAction?: React.ReactNode;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    rowAction?: (params: GridRenderCellParams<any, GQLProductsListManualFragment, any>) => React.ReactNode;
+    hideCrudContextMenu?: boolean;
```

Comment:
Yes, that's what I meant. But we should discuss if this is a good idea first 😅

Reply:
Shouldn't this be readonly instead? As in the generator?

Reply:
this is now solved in #2428 by creating a copy of productsGrid and modifying. I'll postpone the change for this pull-request because I assume we will go with #2428